File: app/src/main/java/com/codekidlabs/storagechooserdemo/MainActivity.java
package com.codekidlabs.storagechooserdemo;

import android.Manifest;
import android.content.pm.PackageManager;
import android.content.res.Configuration;
import android.os.Build;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.Toast;


import com.codekidlabs.storagechooser.ChooserStyle;
import com.codekidlabs.storagechooser.ChooserType;
import com.codekidlabs.storagechooser.Config;
import com.codekidlabs.storagechooser.StorageChooser2;
import com.google.android.material.snackbar.Snackbar;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private CoordinatorLayout coordinatorLayout;
    private CheckBox dirCheckBox, fileCheckBox;

    private String TAG = getClass().getName();

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        if(newConfig.hardKeyboardHidden == Configuration.HARDKEYBOARDHIDDEN_YES) {
            Toast.makeText(getApplicationContext(), "Somethign", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        coordinatorLayout = (CoordinatorLayout) findViewById(R.id.coord);
        dirCheckBox = (CheckBox) findViewById(R.id.checkbox_dir);
        fileCheckBox = (CheckBox) findViewById(R.id.checkbox_file);

//        ((CheckBox) findViewById(R.id.checkbox_membar)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.withMemoryBar(isChecked);
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_free_space_label)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.hideFreeSpaceLabel(isChecked);
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_add_folder)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.allowAddFolder(isChecked);
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_hidden)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.showHidden(isChecked);
//            }
//        });
//
//        dirCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.allowCustomPath(isChecked);
//
//                if (isChecked) {
//                    builder.setType(StorageChooser.DIRECTORY_CHOOSER);
//                    fileCheckBox.setEnabled(false);
//                } else {
//                    fileCheckBox.setEnabled(true);
//                }
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_skip)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.skipOverview(isChecked);
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_session)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.shouldResumeSession(isChecked);
//            }
//        });
//
//        fileCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.allowCustomPath(isChecked);
//
//                if (isChecked) {
//                    builder.setType(StorageChooser.FILE_PICKER);
//                    dirCheckBox.setEnabled(false);
//                } else {
//                    dirCheckBox.setEnabled(true);
//                }
//
//            }
//        });
//
//        ((CheckBox) findViewById(R.id.checkbox_threshold)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.applyMemoryThreshold(isChecked);
//                builder.withThreshold(1, DiskUtil.IN_GB);
//            }
//        });

//        ((CheckBox) findViewById(R.id.checkbox_dark)).setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
//            @Override
//            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
//                builder.setTheme(getScTheme(isChecked));
//            }
//        });
//
//
//        Spinner spinner = (Spinner) findViewById(R.id.filter_spinner);
//        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this,
//                R.array.filter_array, android.R.layout.simple_spinner_item);
//        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
//        spinner.setAdapter(adapter);
//        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
//
//            @Override
//            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
//                switch (i) {
//                    case 0:
//                        builder.crunch();
//                        builder.filter(null);
//                        break;
//                    case 1:
//                        builder.crunch();
//                        builder.filter(StorageChooser.FileType.VIDEO);
//                        break;
//                    case 2:
//                        builder.crunch();
//                        builder.filter(StorageChooser.FileType.AUDIO);
//                        break;
//                    case 3:
//                        builder.crunch();
//                        builder.filter(StorageChooser.FileType.DOCS);
//                        break;
//                    case 4:
//                        builder.crunch();
//                        builder.filter(StorageChooser.FileType.IMAGES);
//                        break;
//                    case 5:
//
//                        ArrayList<String> formats = new ArrayList<>();
//                        formats.add("txt");
//                        formats.add("mkv");
//                        builder.customFilter(formats);
//                        break;
//                }
//            }

//            @Override
//            public void onNothingSelected(AdapterView<?> adapterView) {
//
//            }
//        });


//        // ----------------- Localization -------------------
//        Content c = new Content();
//        c.setCreateLabel("Create");
//        c.setInternalStorageText("My Storage");
//        c.setCancelLabel("Cancel");
//        c.setSelectLabel("Select");
//        c.setOverviewHeading("Choose Drive");
//
//        builder.withActivity(this)
//                .withFragmentManager(getFragmentManager())
//                .setMemoryBarHeight(1.5f)
////                .disableMultiSelect()
//                .withContent(c);


        findViewById(R.id.fab).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                chooser = builder.build();
//                chooser.setOnSelectListener(new StorageChooser.OnSelectListener() {
//                    @Override
//                    public void onSelect(String path) {
//                        Toast.makeText(getApplicationContext(), path, Toast.LENGTH_SHORT).show();
//                    }
//                });
//
//                chooser.setOnCancelListener(new StorageChooser.OnCancelListener() {
//                    @Override
//                    public void onCancel() {
//                        Toast.makeText(getApplicationContext(), "Storage Chooser Cancelled.", Toast.LENGTH_SHORT).show();
//                    }
//                });
//
//                chooser.setOnMultipleSelectListener(new StorageChooser.OnMultipleSelectListener() {
//                    @Override
//                    public void onDone(ArrayList<String> selectedFilePaths) {
//                        for(String s: selectedFilePaths) {
//                            Log.e(TAG, s);
//                        }
//                    }
//                });
//
//                chooser.show();

                Config config = new Config();
                config.setDarkMode(true);
                config.setType(ChooserType.FILE);
                config.setShowMemoryBar(true);
                config.setMemoryBarHeight(1.0f);
                StorageChooser2 chooser2 = new StorageChooser2(getSupportFragmentManager(), config);
                chooser2.show();
            }

        });
    }

//    private StorageChooser.Theme getScTheme(boolean isChecked) {
//        StorageChooser.Theme theme = new StorageChooser.Theme(getApplicationContext());
//        theme.setScheme((isChecked) ? getResources().getIntArray(R.array.paranoid_theme) : theme.getDefaultScheme());
//        return theme;
//    }

    @Override
    protected void onResume() {
        super.onResume();

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && coordinatorLayout != null && ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            Snackbar.make(coordinatorLayout, "Demo app needs storage permission to display file list", Snackbar.LENGTH_INDEFINITE)
                    .setAction("GRANT", new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, 199);
                        }
                    }).show();
        }
    }
}

File: app/src/main/java/com/codekidlabs/storagechooserdemo/MainActivity.kt
package com.codekidlabs.storagechooserdemo

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.google.android.material.snackbar.Snackbar
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.CheckBox
import android.widget.Spinner
import android.widget.Toast

import com.codekidlabs.storagechooser.Content
import com.codekidlabs.storagechooser.OldStorageChooser
import com.codekidlabs.storagechooser.StorageChooser
import com.codekidlabs.storagechooser.utils.DiskUtil

import java.util.ArrayList

class MainActivity : AppCompatActivity() {

    private var coordinatorLayout: CoordinatorLayout? = null
    private var dirCheckBox: CheckBox? = null
    private var fileCheckBox: CheckBox? = null

    private val builder = OldStorageChooser.Builder()
    private var chooser: OldStorageChooser? = null
    private val TAG = javaClass.getName()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        coordinatorLayout = findViewById<View>(R.id.coord) as CoordinatorLayout
        dirCheckBox = findViewById<View>(R.id.checkbox_dir) as CheckBox
        fileCheckBox = findViewById<View>(R.id.checkbox_file) as CheckBox

        (findViewById<View>(R.id.checkbox_membar) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.withMemoryBar(isChecked) }

        (findViewById<View>(R.id.checkbox_free_space_label) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.hideFreeSpaceLabel(isChecked) }

        (findViewById<View>(R.id.checkbox_add_folder) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.allowAddFolder(isChecked) }

        (findViewById<View>(R.id.checkbox_hidden) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.showHidden(isChecked) }

        dirCheckBox!!.setOnCheckedChangeListener { buttonView, isChecked ->
            builder.allowCustomPath(isChecked)

            if (isChecked) {
                builder.setType(OldStorageChooser.DIRECTORY_CHOOSER)
                fileCheckBox!!.isEnabled = false
            } else {
                fileCheckBox!!.isEnabled = true
            }
        }

        (findViewById<View>(R.id.checkbox_skip) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.skipOverview(isChecked) }

        (findViewById<View>(R.id.checkbox_session) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.shouldResumeSession(isChecked) }

        fileCheckBox!!.setOnCheckedChangeListener { buttonView, isChecked ->
            builder.allowCustomPath(isChecked)

            if (isChecked) {
                builder.setType(OldStorageChooser.FILE_PICKER)
                dirCheckBox!!.isEnabled = false
            } else {
                dirCheckBox!!.isEnabled = true
            }
        }

        (findViewById<View>(R.id.checkbox_threshold) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked ->
            builder.applyMemoryThreshold(isChecked)
            builder.withThreshold(1, DiskUtil.IN_GB)
        }

        (findViewById<View>(R.id.checkbox_dark) as CheckBox).setOnCheckedChangeListener { buttonView, isChecked -> builder.setTheme(getScTheme(isChecked)) }


        val spinner = findViewById<View>(R.id.filter_spinner) as Spinner
        val adapter = ArrayAdapter.createFromResource(this,
                R.array.filter_array, android.R.layout.simple_spinner_item)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner.adapter = adapter
        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {

            override fun onItemSelected(adapterView: AdapterView<*>, view: View, i: Int, l: Long) {
                when (i) {
                    0 -> {
                        builder.crunch()
                        builder.filter(null)
                    }
                    1 -> {
                        builder.crunch()
                        builder.filter(OldStorageChooser.FileType.VIDEO)
                    }
                    2 -> {
                        builder.crunch()
                        builder.filter(OldStorageChooser.FileType.AUDIO)
                    }
                    3 -> {
                        builder.crunch()
                        builder.filter(OldStorageChooser.FileType.DOCS)
                    }
                    4 -> {
                        builder.crunch()
                        builder.filter(OldStorageChooser.FileType.IMAGES)
                    }
                    5 -> {

                        val formats = ArrayList<String>()
                        formats.add("txt")
                        formats.add("mkv")
                        builder.customFilter(formats)
                    }
                }
            }

            override fun onNothingSelected(adapterView: AdapterView<*>) {

            }
        }


        // ----------------- Localization -------------------
        val c = Content()
        c.createLabel = "Create"
        c.internalStorageText = "My Storage"
        c.cancelLabel = "Cancel"
        c.selectLabel = "Select"
        c.overviewHeading = "Choose Drive"

        builder.withActivity(this)
                .withFragmentManager(supportFragmentManager)
                .setMemoryBarHeight(1.5f)
                //                .skipOverview(true, "/storage/emulated/0/DCIM/Camera")
                //                .disableMultiSelect()
                .withContent(c)


        findViewById<View>(R.id.fab).setOnClickListener {
            chooser = builder.build()
            chooser!!.setOnSelectListener { path -> Toast.makeText(applicationContext, path, Toast.LENGTH_SHORT).show() }

            chooser!!.setOnCancelListener { Toast.makeText(applicationContext, "Storage Chooser Cancelled.", Toast.LENGTH_SHORT).show() }

            chooser!!.setOnMultipleSelectListener { selectedFilePaths ->
                for (s in selectedFilePaths) {
                    Log.e(TAG, s)
                }
            }

            chooser!!.show()

            //                AlertDialog d = new AlertDialog.Builder(MainActivity.this)
            //                        .setTitle("Also")
            //                        .setMessage("Android P")
            //                        .setPositiveButton("OK", new DialogInterface.OnClickListener() {
            //                            @Override
            //                            public void onClick(DialogInterface dialog, int which) {
            //                                Log.e("Awesome", "Awesome");
            //                            }
            //                        })
            //                        .create();
            //                d.show();
        }
    }

    private fun getScTheme(isChecked: Boolean): OldStorageChooser.Theme {
        val theme = OldStorageChooser.Theme(applicationContext)
        theme.scheme = if (isChecked) resources.getIntArray(R.array.paranoid_theme) else theme.defaultScheme
        return theme
    }

    override fun onResume() {
        super.onResume()

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && coordinatorLayout != null && ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            Snackbar.make(coordinatorLayout!!, "Demo app needs storage permission to display file list", Snackbar.LENGTH_INDEFINITE)
                    .setAction("GRANT") { ActivityCompat.requestPermissions(this@MainActivity, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE), 199) }.show()
        }
    }
}
